build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -97,8 +97,21 @@ class AssetGraph {
     }));
   }
 
-  /// Removes the node representing [id] from the graph.
-  AssetNode _remove(AssetId id) => _nodesByPackage[id.package].remove(id.path);
+  /// Removes the node representing [id] from the graph, and all of its
+  /// `primaryOutput`s.
+  ///
+  /// Returns a [Set<AssetId>] of all removed nodes.
+  Set<AssetId> _remove(AssetId id, {Set<AssetId> removedSet}) {
```

Comment:
Could we get away without communicating through an argument if we make this return Iterable and use `sync*`?

Reply:
If we return an iterable then this won't actually run unless somebody reads from the iterable.